python/test/unit/fem/test_petsc_transfer_matrix.py

```diff
@@ -24,21 +24,20 @@ def test_scalar_p1():
     uc = interpolate(u, Vc)
     uf = interpolate(u, Vf)
 
-    mat = PETScDMCollection.create_transfer_matrix(Vc, Vf)
+    mat = PETScDMCollection.create_transfer_matrix(Vc, Vf).mat()
     Vuc = Function(Vf)
-    mat.mult(uc.vector(), Vuc.vector())
-    Vuc.vector().update_ghost_values()
+    mat.mult(uc.vector().vec(), Vuc.vector().vec())
 
-    diff = Function(Vf)
-    diff.assign(Vuc - uf)
-    assert diff.vector().norm("l2") < 1.0e-12
+    diff = Vuc.vector()
+    diff.vec().axpy(-1, uf.vector().vec())
+
+    assert diff.norm(Norm.l2) < 1.0e-12
 
 
-@pytest.mark.xfail
 def test_scalar_p1_scaled_mesh():
     # Make coarse mesh smaller than fine mesh
     meshc = UnitCubeMesh(MPI.comm_world, 2, 2, 2)
-    for x in meshc.coordinates():
+    for x in meshc.geometry.points:
```

Comment:
Apparently not:
```
>       meshc.geometry.points *= 0.9
E       AttributeError: can't set attribute
```

Reply:
`meshc.geometry.points` is set to read-only in the pybind11 layer. Could change it. Adding a value via a loop in Python is a bit untidy (and slow).

Reply:
OK, but it's not what we're testing here. If you want to change to read-write then let's make another PR.

Reply:
It should be fixed now because this type of code can made tests very slow, plus users will model their own codes on it. I made the XDMF over 10 times faster by fixing this sort poor implementation.

Reply:
I've started a new branch to look at this, and any other pybind11 tweaks.